### ISSUE TITLE
errors installing pycsep from --channel conda-forge

### ISSUE THREAD

**Issue by pjmaechling** [SCECcode/pycsep#41]
I tried installing pycsep from the conda --channel conda-forge on USC's new cluster discovery. I started by installing the latest Anaconda3 distribution, which seemed to install correctly.

(base) [maechlin@discovery ~]$ conda info

     active environment : base
    active env location : /home1/maechlin/anaconda3
            shell level : 1
       user config file : /home1/maechlin/.condarc
 populated config files : 
          conda version : 4.8.5
    conda-build version : 3.18.11
         python version : 3.8.3.final.0
       virtual packages : __glibc=2.17
       base environment : /home1/maechlin/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home1/maechlin/anaconda3/pkgs
                          /home1/maechlin/.conda/pkgs
       envs directories : /home1/maechlin/anaconda3/envs
                          /home1/maechlin/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.5 requests/2.24.0 CPython/3.8.3 Linux/3.10.0-1062.4.1.el7.x86_64 centos/7.7.1908 glibc/2.17
                UID:GID : 14364:30916
             netrc file : None
           offline mode : False

(base) [maechlin@discovery ~]$ conda env list
# conda environments:
#
base         * /home1/maechlin/anaconda3

When I tried to install pycsep from the conda-forge channel, the process worked for a long time, then output an error report that starts with the following information. I have not include the full error report, because it is quite lengthy. Any suggestions on why this install didn't work using Anaconda3? 

(base) [maechlin@discovery ~]$ conda install --channel conda-forge pycsep
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes. Press CTRL-C to abort.
Examining conflict for jinja2 anaconda conda-build jupyterlab bokeh jupyterlab_server conda-verify sphinx markupsafe notebook nbconvExamining conflict for jinja2 anaconda conda-build jupyterlab bokeh jupyterlab_server conda-verify sphinx markupsafe notebook nbconvExamining conflict for anaconda xlsxwriter: : 327it [50:36, 2.11s/it]                                                             Examining conflict for fontconfig lxml dbus pyodbc pango glib conda-build harfbuzz libarchive gst-plugins-base python-libarchive-c cExamining conflict for fontconfig lxml dbus pyodbc pango glib conda-build harfbuzz libarchive gst-plugins-base python-libarchive-c cExamining conflict for qt cairo: : 342it [50:51, 1.65s/it]                                                                     failed                                                                                                                           
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:
Specifications:
 - alabaster -> python[version=’2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|3.4.*|>=3.7,<3.8.0a0’]
 - anaconda-navigator -> python[version=’2.7.*|3.5.*|3.6.*|3.4.*’]
 - anaconda-project -> python[version=’2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.7,<3.8.0a0|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0']
 - atomicwrites -> python[version=’2.7.*|3.5.*|3.6.*|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0']
 - attrs -> python[version=’2.7.*|3.5.*|3.6.*|>=3.7,<3.8.0a0|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0']
 - autopep8 -> python[version=’2.7.*|3.5.*|3.6.*|3.4.*|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0']

**Comment by wsavran** [comment]
thanks phil. do you mind trying the installation inside of a virtual environment? anaconda comes with a bunch of preinstalled packages that can cause dependency issues, which is why we recommend miniconda. can you send the full error report? maybe we can pin some requirements in the next conda-forge feedstock to prevent this issue from happening in the future.

`conda create --name pycsep python=3.7`
`conda activate pycsep`
`conda install --channel conda-forge pycsep`

**Comment by wsavran** [comment]
to follow up, i was able to install on discovery using miniconda. it looks like there is some package conflict that cannot be resolved. the conda-forge feedstock recipe does not pin any requirements, so if you can send over the full error report i shoudl be able to add these pins to prevent this from happening.

**Comment by pjmaechling** [comment]
I used my anaconda3 environment and created a virtual environment like you gave above, but used python=3.8 and the installation worked for me on discovery. I was able to run a small program that imported pycsep. The errorr log when installing pycsep using the standard Anaconda3 distribution is attached below. I think this issue can be closed. Thanks, Phil M.
[pycsep_install_errs.docx](https://github.com/SCECcode/pycsep/files/5273246/pycsep_install_errs.docx)
